fog-azure-rm.gemspec

```diff
@@ -32,7 +32,7 @@ Gem::Specification.new do |spec|
   spec.add_dependency 'azure_mgmt_traffic_manager', '~> 0.9.0'
   spec.add_dependency 'azure_mgmt_sql', '~> 0.9.0'
   spec.add_dependency 'azure_mgmt_key_vault', '~> 0.9.0'
-  spec.add_dependency 'azure-storage', '= 0.11.5.preview'
+  spec.add_dependency 'azure-storage', '~> 0.11.preview'
```

Comment:
Can we use the latest version `0.15.0.preview` here?

Reply:
@bguban, `~> 0.11.preview` will take `0.15.0.preview` too.

Reply:
`~> 0.11.preview` means `>= 0.11` including preview and `< 1.0`

Reply:
Can you explain how it will take 0.15.0.preview as well? When I install the gem with these dependencies, it just shows as `0.11.preivew`

Reply:
According to [thoughtbot](https://thoughtbot.com/blog/rubys-pessimistic-operator) (specifically the second part - ~> 1.1 example) this will result with `>= 0.11` and ` < 1.0` as I have mentioned before. According to [rubygems pattern guide](https://guides.rubygems.org/patterns/)

> Using ~> with prerelease versions will restrict you to prerelease versions only.

Which should be good, since Azure only releases `.preview` versions to [rubygems](https://rubygems.org/gems/azure-storage)

We already tested it, and `0.15.preview` was pulled.

Reply:
@maham-nazir-confiz, did you run `bundle update`? Perhaps you have `Gemfile.lock` with `0.11.preview` locked. `bundle update` should update the version to `0.15.preview`, as last resort you can try and delete `Gemfile.lock` completely and run `bundle update`.

Reply:
yeah I got it to work. so why not use the 0.15 directly in gemspec?

Reply:
I wanted a minimal impact on the original gemspec, so I left an option to roll back to `0.11.5` if `0.15` was causing a problem.
I suggest leaving it with `~> 0.11`, until there will be a breaking change in `azure-storage` (which shouldn't happen, since such changes should advance the major version, if it is versioned correctly)

Reply:
The latest `0.15.preview` is actually causing a breaking change. It has a dependency on `nokogiri =1.6.8, ~> 1.6`. It tries to install the latest nokogiril version which is `1.10.3` which depends on `Ruby-2.3` or higher

Reply:
@maham-nazir-confiz, @ShimShtein maybe as possible solution is just to increment the major version of the `fog-azure-rm` and use the latest versions of dependent gems in it. If somebody needs backward compatibility they would use the current version, who don't need it could use the newer version of the gem. 
For now, my project's security checker shows me that I have an old `nokogiri` version which has critical security problems. The politic of my company forces me to solve this problem (and I think I'm not alone in it).
So what do you think guys?

Reply:
> @maham-nazir-confiz, @ShimShtein maybe as possible solution is just to increment the major version of the `fog-azure-rm` and use the latest versions of dependent gems in it. If somebody needs backward compatibility they would use the current version, who don't need it could use the newer version of the gem.

I agree, upgrading to later gem versions would resolve much of these issues.

Reply:
@bguban This can be a possible solution.
@maham-nazir-confiz your thoughts on @bguban 's comment?

Reply:
@maham-nazir-confiz, Actually it depends on the user to limit `nokogiri` version. I mean the user can limit `nokogiri` version in his own Gemspec file like this:

``` ruby
gem 'fog-azure-rm' # require our library
gem 'nokogiri', '<=1.9.1' # if the user wants ruby 2.0 compatibility
```
`azure-storage` does not require newer 'nokogiri', it sets only lower limit requirement. It's up to user to set up higher limit, if the user needs it.

Other option would be adding an `if` condition into `fog-azure-rm.gemspec`, but I am not sure it will be parsed well by `rubygems`.

Reply:
@ShimShtein That's right, users can set the upper limit of nokogiri, if required, in their own gemspec. 
@maham-nazir-confiz What do you think?

Reply:
@maham-nazir-confiz  Could you please let us know if the approach suggested against nokogiri failure works for you?